src/PowerShellEditorServices/Workspace/Workspace.cs

```diff
@@ -202,6 +213,35 @@ public string GetRelativePath(string filePath)
             return resolvedPath;
         }
 
+        internal static string MapRemotePathToLocal(string remotePath, string connectionString)
+        {
+            string sessionTempPath =
+                Path.Combine(
+                    Path.GetTempPath(),
+                    "PSES-" + Process.GetCurrentProcess().Id,
+                    "RemoteFiles");
+
+            // The path generated by this code will look something like
+            // %TEMP%\PSES-[PID]\RemoteFiles\1205823508\computer-name\MyFile.ps1
```

Comment:
This is a hash of the file's folder path (sans filename) on the remote machine so a file will always end up with the same "slug" (I should use a better name) when loaded in the future so long as its path remains the same.